hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/fsck/ContainerHealthTask.java

```diff
@@ -0,0 +1,345 @@
+    return new ContainerHealthStatus(container, replicas, placementPolicy);
+  }
+
+  private void completeProcessingContainer(ContainerHealthStatus container,
+      Set<String> existingRecords, long currentTime) {
+    if (container != null) {
+      containerSchemaManager.insertUnhealthyContainerRecords(
+          ContainerHealthRecords.generateUnhealthyRecords(
+              container, existingRecords, currentTime));
+      processedContainers.add(container.getContainer());
+    }
+  }
+
+  private long processExistingDBRecords(long currentTime) {
+    long recordCount = 0;
+    try (Cursor<UnhealthyContainersRecord> cursor =
+             containerSchemaManager.getAllUnhealthyRecordsCursor()) {
+      ContainerHealthStatus currentContainer = null;
+      Set<String> existingRecords = new HashSet<>();
+      while(cursor.hasNext()) {
+        recordCount++;
+        UnhealthyContainersRecord rec = cursor.fetchNext();
+        try {
+          if (currentContainer == null
+              || currentContainer.getContainerID() != rec.getContainerId()) {
+            completeProcessingContainer(
+                currentContainer, existingRecords, currentTime);
+            existingRecords.clear();
+            currentContainer = setCurrentContainer(rec.getContainerId());
+          }
+          if (ContainerHealthRecords
+              .retainOrUpdateRecord(currentContainer, rec)) {
+            existingRecords.add(rec.getContainerState());
+            if (rec.changed()) {
+              rec.update();
+            }
+          } else {
+            rec.delete();
+          }
+        } catch (ContainerNotFoundException cnf) {
+          rec.delete();
+          currentContainer = null;
+        }
+      }
+      // Remember to finish processing the last container
+      completeProcessingContainer(
+          currentContainer, existingRecords, currentTime);
+    }
+    return recordCount;
+  }
+
+  private void processContainer(ContainerInfo container, long currentTime) {
+    try {
+      if (processedContainers.contains(container)) {
+        // Was already handled when processing the existing DB records, so skip
+        return;
+      }
+      Set<ContainerReplica> containerReplicas =
+          containerManager.getContainerReplicas(container.containerID());
+      ContainerHealthStatus h = new ContainerHealthStatus(
+          container, containerReplicas, placementPolicy);
+      if (h.isHealthy()) {
+        return;
+      }
+      containerSchemaManager.insertUnhealthyContainerRecords(
+          ContainerHealthRecords.generateUnhealthyRecords(h, currentTime));
+    } catch (ContainerNotFoundException e) {
+      LOG.error("Container not found while finding missing containers", e);
+    }
+  }
+
+  /**
+   * Helper methods to generate and update the required database records for
+   * unhealthy containers.
+   */
+  static public class ContainerHealthRecords {
+
+    static public boolean retainOrUpdateRecord(
```

Comment:
Can you please add JavaDoc for this method?

Reply:
Done.